packages/python-google-compute-engine/google_compute_engine/metadata_scripts/script_retriever.py

```diff
@@ -22,12 +22,28 @@
 import tempfile
 
 from google_compute_engine import metadata_watcher
+from google_compute_engine import service_utils
 from google_compute_engine.compat import httpclient
 from google_compute_engine.compat import urlerror
 from google_compute_engine.compat import urlrequest
 from google_compute_engine.compat import urlretrieve
 
 
+def RetryOnUnavailable(func):
+  """Function decorator to retry on a service unavailable exception."""
+  return service_utils.RetryOnUnavailable(func, max_retry=3)
+
+
+@RetryOnUnavailable
+def UrlOpenWithRetry(request):
```

Comment:
All functions should have a doc comment with argument descriptions.

Reply:
done, and started with "_" to mark it as internal